src/machines/confirmation-dialog-machine.ts

```diff
@@ -0,0 +1,169 @@
+// Source: https://xstate-catalogue.com/machines/confirmation-dialog
```

Comment:
The machine.

![](https://media1.giphy.com/media/Rm8vFJZ6eaUU/200.gif?cid=5a38a5a23xx0stohc3j024ui49oj9ytax9r5ynidjbw3v1bt&rid=200.gif&ct=g)

This state machine is an expanded/tweaked version of the [Confirmation Dialog machine from XState Catalogue](https://xstate-catalogue.com/machines/confirmation-dialog).

The primary paths are:

1. `Closed -(Open Dialog)-> Open (Idle) -(Cancel)-> Closed`
2. `Closed -(Open Dialog)-> Open (Idle) -(Confirm)-> Open (Execute Action) -(onDone)-> Closed`

See [the graph at the top of this page](https://xstate-catalogue.com/machines/confirmation-dialog) for exhaustive details.

If the `Confirm` transition is triggered. The machine will make an API request to attempt to remove the member from the team.

If that request succeeds, it will go down the `onDone` path which closes the dialog, updates the list of members to be displayed, and triggers a success toast.

If that request fails, it will go down the `onError` path which closes the dialog, leaves the list of members as is, and triggers a failure toast.